File: ios/ViewManagers/SmileIDDocumentCaptureViewManager.swift
import Foundation
import React
import SwiftUI

@objc(SmileIDDocumentCaptureViewManager)
class SmileIDDocumentCaptureViewManager: SmileIDBaseViewManager {
    override func getView() -> UIView {
        BaseSmileIDView(frame: .zero, contentView: AnyView(SmileIDDocumentCaptureView(product: product, smileIDUIViewDelegate: self)), product: product)
    }

    @objc func setParams(_ node: NSNumber, commandId _: NSNumber, params: NSDictionary) {
        /*  UI Updates on the Main Thread:async ensures that the UI update is scheduled to run on the next cycle of the run loop, preventing any potential blocking of the UI if the update were to take a noticeable amount of time
         */
        DispatchQueue.main.async {
            if let component = self.bridge.uiManager.view(forReactTag: node) as? BaseSmileIDView {
              
                var autoCapture: String?
                if let autoCaptureValue = params["autoCapture"] as? String {
                  autoCapture = autoCaptureValue
                }
              
                self.product.extraPartnerParams = params["extraPartnerParams"] as? [String: String] ?? [:]
                self.product.userId = params["userId"] as? String
                self.product.jobId = params["jobId"] as? String
                self.product.front = params["isDocumentFrontSide"] as? Bool ?? true
                self.product.showAttribution = params["showAttribution"] as? Bool ?? true
                self.product.showInstructions = params["showInstructions"] as? Bool ?? true
                self.product.showConfirmation = params["showConfirmation"] as? Bool ?? true
                self.product.allowGalleryUpload = params["allowGalleryUpload"] as? Bool ?? false
                self.product.autoCaptureTimeout = params["autoCaptureTimeout"] as? Int ?? 10
                self.product.autoCapture = autoCapture?.toAutoCapture() ?? .autoCapture
                self.product.onResult = params["onResult"] as? RCTBubblingEventBlock
            }
        }
    }
}

File: android/src/main/java/com/smileidentity/react/viewmanagers/SmileIDDocumentCaptureViewManager.kt
package com.smileidentity.react.viewmanagers

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.smileidentity.react.toAutoCapture
import com.smileidentity.react.utils.getBoolOrDefault
import com.smileidentity.react.utils.getIntOrDefault
import com.smileidentity.react.utils.getStringOrDefault
import com.smileidentity.react.views.SmileIDDocumentCaptureView
import com.smileidentity.react.views.SmileIDDocumentVerificationView

@ReactModule(name = SmileIDDocumentCaptureViewManager.NAME)
class SmileIDDocumentCaptureViewManager(
  private val reactApplicationContext: ReactApplicationContext
) : BaseSmileIDViewManager<SmileIDDocumentCaptureView>(reactApplicationContext) {
  override fun getName(): String = NAME

  override fun createSmileView(): SmileIDDocumentCaptureView {
    return SmileIDDocumentCaptureView(reactApplicationContext)
  }

  override fun applyArgs(view: SmileIDDocumentCaptureView, args: ReadableMap?) {
    args?.let {
      view.userId = it.getStringOrDefault("userId")
      view.jobId = it.getStringOrDefault("jobId")
      view.showAttribution = it.getBoolOrDefault("showAttribution", true)
      view.showInstructions = it.getBoolOrDefault("showInstructions", true)
      view.showConfirmation = it.getBoolOrDefault("showConfirmation", true)
      view.allowGalleryUpload = it.getBoolOrDefault("allowGalleryUpload", false)
      view.front = it.getBoolOrDefault("isDocumentFrontSide", true)
      view.autoCaptureTimeout = it.getIntOrDefault("autoCaptureTimeout", null)
      view.autoCapture = it.getStringOrDefault("autoCapture", null)?.toAutoCapture()
    }
  }

  companion object {
    const val NAME = "SmileIDDocumentCaptureView"
  }
}
